tests: net: socket: register: move to new ztest API

hunterhu3000 opened the pull request:
Move net/socket/register tests to use new ztest API
TEST=twister -T tests/net/socket/register

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/register/src/main.c:
```diff
@@ -223,7 +223,7 @@ NET_SOCKET_REGISTER(tls,       TEST_SOCKET_PRIO, AF_UNSPEC, is_tls,    socket_te
 NET_SOCKET_REGISTER(af_can,    TEST_SOCKET_PRIO, AF_CAN,    is_can,    socket_test_ok);
 
-void test_create_sockets(void)
+ZTEST(socket_register_test_suite, test_create_sockets)
```
no need for _test_suite here.

hunterhu3000 replied:
Updated. Thanks to review.

hunterhu3000 committed:
tests: net: socket: register: move to new ztest API

Move net/socket/register tests to use new ztest API
TEST=twister -T tests/net/socket/register

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case should be run with specific order.
